### PR TITLE
[Fixes #388] ensure broccoli-cli exits with the appropriate status co…

### LINKED ISSUE #388: Broccoli build returns 0 on build failure
When plugins fail throwing errors, `broccoli build` still returns exit code 0, despite the fact that it has not built successfully, and has not created the build target directory.

### PR DESCRIPTION
**Pull request by stefanpenner**
…de on build failure

### CODE REVIEW

File: lib/cli.ts [hunk 1]
```diff
@@ -173,6 +176,9 @@ export = function broccoliCLI(args: string[], ui = new UI()) {
       watcher.on('buildFailure', (err: any) => {
         ui.writeLine('build failure', 'ERROR');
         ui.writeError(err);
+        if (!options.watch) {
+          process.exitCode = 1;
```
Comment (stefanpenner) [inline lib/cli.ts]: todo: test this scenario

**Comment by peabnuts123** [comment]
Interesting. How does calling `process.exitz(1)` mean the process exits with the wrong code? I frequently do this in my own code. What's the benefit of setting `process.exitCode` first?

**Comment by stefanpenner** [comment]
It doesn’t, it’s just a more cooperative way to shut down. Specifically the bug fix here was adding a missing exitCode assignment (see above inline comment)

**Comment by peabnuts123** [comment]
Oh, I see! I had missed that there was an new exit scenario in there.

**Comment by ryuran** [comment]
Something new for this PR @stefanpenner @rwjblue ?

**Comment by stefanpenner** [comment]
released as v3.5.1 🎉

**Pull request closed without merging**